components/src/components/dateRangeSelector/date-range-selector.ts

```diff
@@ -155,14 +147,45 @@ export class DateRangeSelector extends TailwindElement() {
                     type="text"
                     placeholder="Date from"
                     id="date-picker"
+                    @change=${(event: Event) => {
+                        event.preventDefault();
+                        this.dateToPicker?.set('minDate', this.dateFromPicker?.selectedDates[0]);
+                        this.selectedValue = 'custom';
+                    }}
                 />
                 <input
                     class="input input-bordered rounded-none join-item"
                     type="text"
                     placeholder="Date to"
                     id="end-date-picker"
+                    @change=${(event: Event) => {
+                        event.preventDefault();
+                        this.dateFromPicker?.set('maxDate', this.dateToPicker?.selectedDates[0]);
+                        this.selectedValue = 'custom';
+                    }}
                 />
-            </div>
+                <button
+                    class="btn btn-primary join-item"
+                    @click="${(event: PointerEvent) => {
+                        event.preventDefault();
+                        this.submit();
+                    }}"
+                >
+                    Submit
+                </button>
+            </form>
         `;
     }
+
+    private submit() {
+        const dateFrom = this.dateFromPicker?.selectedDates[0];
+        const dateTo = this.dateToPicker?.selectedDates[0];
+
+        this.dispatchEvent(
+            new CustomEvent('gs-date-range-changed', {
+                detail: { dateFrom, dateTo },
+                bubbles: true,
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Event/composed

```suggestion
                bubbles: true,
                composed: true,
```